wcfsetup/install/files/lib/data/attachment/Attachment.class.php

```diff
@@ -140,10 +141,44 @@ public function getTinyThumbnailLocation() {
 	 */
 	public function getThumbnailLocation($size = '') {
 		if ($size == 'tiny') {
-			return self::getStorage() . substr($this->fileHash, 0, 2) . '/' . $this->attachmentID . '-tiny-' . $this->fileHash;
+			$location = self::getStorage() . substr($this->fileHash, 0, 2) . '/' . $this->attachmentID . '-tiny-' . $this->fileHash;
+		}
+		else {
+			$location = self::getStorage() . substr($this->fileHash, 0, 2) . '/' . $this->attachmentID . '-thumbnail-' . $this->fileHash;
+		}
+		
+		return $this->getLocationHelper($location);
+	}
+	
+	/**
+	 * Migrates the storage location of this attachment.
+	 */
+	public function migrateStorage() {
+		foreach ([$this->getLocation(), $this->getThumbnailLocation(), $this->getThumbnailLocation('tiny')] as $location) {
+			if (!StringUtil::endsWith($location, '.bin')) {
+				rename($location, $location.'.bin');
+			}
+		}
+	}
+	
+	/**
+	 * Returns the appropriate location with or without extension.
+	 * 
+	 * @param	string $location
+	 * @return	string
+	 */
+	protected final function getLocationHelper($location) {
+		// Check location with extension
```

Comment:
I would change this comment and get rid of the one before the `else if`.
```
// Files are suffixed with `.bin` since 5.2, but they are recognized without the extension for backward compatibility.
```
This explains _why_ rather than _what_ the code is doing.

Reply:
I moved this into the PhpDoc comment.